src/core/push/safari.ts

```diff
@@ -46,7 +46,43 @@ export default class SafariPushManager implements PushOpsManager {
         return result;
     }
 
+    async attemptPushRegister(ctx: Context): Promise<void> {
+        const wasUnregistered = await get<boolean>('wasUnregistered');
+
+        if (wasUnregistered) {
+            console.info('Was unregistered before, not calling push register');
+            return;
+        }
+
+        return this.pushRegister(ctx);
+    }
+
+    async pushUnregister(ctx: Context): Promise<void> {
+        await set<boolean>('wasUnregistered', true);
+
+        const perm = window.safari?.pushNotification.permission(
+            this.safariPushId as string
+        );
+
+        if (!perm || !perm.deviceToken) {
+            return;
+        }
+
+        const existingTokenHash = await get<number>('pushTokenHash');
+        const tokenHash = hashToken(ctx, perm.deviceToken);
+
+        if (existingTokenHash === tokenHash) {
+            return;
+        }
+
+        await trackEvent(ctx, EventType.PUSH_UNREGISTERED);
+
+        await set('pushTokenHash', tokenHash);
```

Comment:
Is it necessary for us to store token hashes during unregister?

Perhaps we should just remove / leave these keys alone?

Reply:
The whole method was a mess. Wasn't tested due to old Safari. Should be fixed now